doc/design/design.md

```diff
@@ -123,204 +142,360 @@ class SelectParameter{
 
-That implies that when doing updates, the extension-manager must check if parameter definitions changed. If there were breaking changes the update can't be performed automatically.
-The only option would be to add update scripts that define how to convert the parameters from one version to another.
-However, that is currently out of scope.
+That implies that when doing updates, the extension-manager must check if parameter definitions changed. If there were breaking changes, the update can't be performed automatically.
 
-### Conditional Parameters
+The only option would be to add update scripts that define how to convert the parameters from one version to another. However, that is currently out of scope.
 
-We need to have conditional parameters. For example for the s3-document-files-virtual-schema we need to provide a select
-for choosing the source type (AWS S3 / Google Cloud Storage / ...). Depending on the source type we want to show / hide
-some parameters. For example the S3 Bucket name.
+#### Conditional Parameters
 
-To solve this, we considered the following options:
+We need to have conditional parameters. An example use case for this is when a virtual schema for another database supports multiple connection protocols or libraries (e.g. ODBC and JDBC) that require different configuration options. Depending on the selected protocol we want to show or hide some parameters.
 
-* Condition as JS code in a string
-  Example: `condition: "parameters.source === \"s3\""`
-* Condition as JS callback. For example:
-   ```js
-  extension = {
-    condition: (parameterValues) => parameterValues.source === "s3"
+We decided to represent the condition as a JSON structure.
+
+Example:
+
+```js
+extension = {
+  // ...
+  condition: {
+    parameter: "connectorType",
+    operator: Operators.EQ,
+    value: "jdbc"
   }
-  ```
-* Condition represented using structured JSON. For example:
+  // ...
+}
+```
+
+While this requires interpretation of the JSON structure, it allows us to transfer the parameter definition to the frontend as simple JSON in contrast to the alternative options considered.
+
+##### Alternative Options
+
+We considered the following options:
+
+* Condition as JavaScript code in a string
+
+  Example: `condition: "parameters.connectorType === \"jdbc\""`
+
+* Condition as JavaScript callback
+
+  Example:
    ```js
   extension = {
-    condition: {
-     parameter: "source",
-     operator: Operators.EQ,
-     value: "s3"
-    }
+    condition: (parameterValues) => parameterValues.connectorType === "jdbc"
   }
   ```
 
-We decided for the third option since it does not require evaluating source code from an HTTP request the client. That
-allows us to transfer the parameter definition as simple JSON (in contrast to a JS lib / file).
+While both options don't need any interpretation logic they would require executing the JavaScript snippet in the frontend with `eval()` which could be interpreted as a security issue.
```

Comment:
```suggestion
While both options don't need any interpretation logic they would require executing the JavaScript snippet in the frontend with `eval()` which poses a security risk.
```
Don't downplay it. We know that it would be a bad idea. :-)